### PR TITLE
Switch default minimum supported TLS version to v1.2

### PR DESCRIPTION
**Pull request by mrueg**
This PR switches the default minimum to TLSv1.2 as recommended in https://blog.pcisecuritystandards.org/migrating-from-ssl-and-early-tls

### CODE REVIEW

**Comment by dciancu** [comment]
+1 for adding TLSv1.3 to supported protocols in Dovecot code, although openssl supports it, we cannot use it due to the check in Dovecot code.

File: src/config/old-set-parser.c [hunk 1]
```diff
@@ -171,7 +171,7 @@ static int ssl_protocols_to_min_protocol(const char *ssl_protocols,
 					 const char **error_r)
 {
 	static const char *protocol_versions[] = {
-		"SSLv3", "TLSv1", "TLSv1.1", "TLSv1.2",
+		"SSLv3", "TLSv1", "TLSv1.1", "TLSv1.2", "TLSv1.3",
```
Comment (darix) [inline src/config/old-set-parser.c:174]: This change alone is not enough to allow "TLSv1.3" only servers.

Comment (mrueg) [reply src/config/old-set-parser.c:174]: Yes it doesn't add TLSv1.3 per se. I looked through the code, config and documentation for occurrences that did not include it and added it there.

Comment (darix) [reply src/config/old-set-parser.c:174]: https://github.com/dovecot/core/pull/126

**Comment by cmouse** [comment]
Fixed via 096b7122b9068ed34224a52c067437107f6f72ab

**Pull request closed without merging**